generators/base/generator-base.mjs

```diff
@@ -2446,7 +2448,10 @@ templates: ${JSON.stringify(existingTemplates, null, 2)}`;
     dest.buildToolGradle = dest.buildTool === GRADLE;
     dest.buildToolMaven = dest.buildTool === MAVEN;
     dest.buildToolUnknown = !dest.buildToolGradle && !dest.buildToolMaven;
-    dest.buildDir = this.getBuildDirectoryForBuildTool(dest.buildTool);
+
+    dest.temporaryDir = dest.buildToolGradle ? 'build/' : 'target/';
```

Comment:
leaving the logic in a dedicated method (ideally in a /support package) seems to be a good idea

Reply:
The entire loadDerivedServerConfig should be moved to generators/server/support in another PR.